chore: separate journal scratch commands

hikchoi opened the pull request:
# chore: separate journal scratch commands
This PR:
- Registers separate commands `Dendron: Lookup Journal Note` and `Dendron: Lookup Scratch Note`

# Pull Request Checklist

If this is your first time submitting a pull request to Dendron, copy and paste the full [Dendron Review Checklist](https://docs.dendron.so/notes/1EoNIXzgmhgagqcAo9nDn.html) into this request and check off each item as necessary. 

This template contains the short checklist which is used by the Dendron core team. 

## Testing
- [~] [Write Tests](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#writing-tests) 
- [x] [Confirm existing tests pass](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#executing-tests)
- [x] [Confirm manual testing](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#manual-testing) 
- [~] If your tests changes an existing snapshot, make sure that snapshots are [updated](https://docs.dendron.so/notes/99q7A73uGmCwu2KvSHZro.html#updating-test-snapshots)
- [~] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in the [test workspace](https://docs.dendron.so/notes/dtMsF12SF2SUhLN10sYe2.html)

### Docs
- [x] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://docs.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

hikchoi commented:
madge reports no additional circular dependencies in `plugin-core` compared to master

hikchoi commented:
docs: https://github.com/dendronhq/dendron-site/pull/381

jonathanyeung reviewed (changes requested):
Can we add a quick test for each new command - basically invoke the command, check that lookup got invoked with the noteType:journal/scratch parameter?

Added 1 minor comment on the docs review, otherwise lgtm

hikchoi commented:
Update: 
- cleaned up docs
- rebased branch to latest master
- added tests for correct invocation
  - as part of adding the tests, I've migrated `openNote` into `WSUtilsV2`

Madge reports no additional circular dependencies compared to master
